test/inspect.test.ts

```diff
@@ -0,0 +1,272 @@
+import * as fs from 'fs';
+import * as tap from 'tap';
+import * as _ from 'lodash';
+import * as path from 'path';
+import request from 'sync-request';
+
+import composerParser = require('../lib');
+import systemVersionsStub = require('./stubs/system_deps_stub');
+
+const deepTestFolders = [
+    'proj_with_no_deps',
+    'vulnerable_project',
+    'circular_deps_php_project',
+    'many_deps_php_project',
+    'circular_deps_special_test',
+    'proj_with_aliases',
+    'proj_with_aliases_external_github',
+    'no_branch_alias',
+];
+
+deepTestFolders.forEach((folder) => {
+    tap.test('php plugin for ' + folder, async (t) => {
+        const projFolder = './test/fixtures/' + folder;
+
+        try {
+            const depTree = composerParser.buildDepTreeFromFiles(projFolder, 'composer.lock', systemVersionsStub);
+
+            t.test('match packages with expected', (test) => {
```

Comment:
I think if you use `await t.test('...', async (test) => {...` - then you won't need t.end() inside.
But I'm not sure. Tap is difficult.

Reply:
That didn't work. Tap is more strange then difficult